lib/github/kv.rb

```diff
@@ -252,6 +253,32 @@ def setnx(key, value, expires: nil)
       }
     end
 
+    # increment :: String, Integer, expires: Time? -> Integer
+    #
+    # Increment the key's value by an amount.
+    #
+    # key     - The key to increment.
+    # amount  - The amount to increment the key's value by.
+    # expires - When the key should expire.
+    #
+    # Returns the key's value after incrementing.
+    def increment(key, amount: 1, expires: nil)
+      sql = GitHub::SQL.run(<<-SQL, key: key, amount: amount, now: now, expires: expires || GitHub::SQL::NULL, connection: connection)
+        INSERT INTO key_values (`key`, value, created_at, updated_at, expires_at)
+        VALUES(:key, :amount, :now, :now, :expires)
+        ON DUPLICATE KEY UPDATE
+          `value`=LAST_INSERT_ID(IF(
```

Comment:
`LAST_INSERT_ID` is a hack here to allow us to "return" the new value in a single query. Since an update is being performed we're able to manually set the last insert ID to be any value we wish, which in this case is the new value.